modules/perception/camera/lib/obstacle/detector/yolo/region_output.cu

```diff
@@ -476,35 +519,57 @@ void get_objects_gpu(const YoloBlobs &yolo_blobs,
   const float *rtvis_data = get_gpu_data(with_lights, *yolo_blobs.rtvis_blob);
   const float *rtswt_data = get_gpu_data(with_lights, *yolo_blobs.rtswt_blob);
 
+  int all_scales_num_candidates = 0; 
+  for (size_t i = 0; i < num_candidates_vec.size(); i++){
+    all_scales_num_candidates += num_candidates_vec[i];
+  }
   yolo_blobs.res_box_blob->Reshape(
-      std::vector<int>{1, 1, num_candidates, kBoxBlockSize});
+      std::vector<int>{1, 1, all_scales_num_candidates, kBoxBlockSize});
   yolo_blobs.res_cls_blob->Reshape(
-      std::vector<int>{1, 1, num_classes + 1, num_candidates});
+      std::vector<int>{1, 1, num_classes + 1, all_scales_num_candidates});
 
   float *res_box_data = yolo_blobs.res_box_blob->mutable_gpu_data();
   float *res_cls_data = yolo_blobs.res_cls_blob->mutable_gpu_data();
   const int thread_size = 512;
-
-  int block_size = (num_candidates + thread_size - 1) / thread_size;
-  get_object_kernel <<< block_size, thread_size, 0, stream >>> (
-          num_candidates, loc_data, obj_data, cls_data, ori_data, dim_data,
-          lof_data, lor_data, area_id_data,
-          visible_ratio_data, cut_off_ratio_data,
-          brvis_data, brswt_data, ltvis_data, ltswt_data,
-          rtvis_data, rtswt_data,
-          yolo_blobs.anchor_blob->gpu_data(),
-          yolo_blobs.expand_blob->gpu_data(),
-          width, height,
-          num_anchor, num_classes, model_param.confidence_threshold(),
-          light_vis_conf_threshold, light_swt_conf_threshold,
-          model_param.with_box3d(), model_param.with_frbox(),
-          model_param.with_lights(), model_param.with_ratios(),
-          model_param.num_areas(),
-          res_box_data, res_cls_data);
-  cudaStreamSynchronize(stream);
+  //TODO[KaWai]: use different stream to process scales in parallel.
+  int num_candidates_offset = 0;
+  for (int i = 0; i < num_candidates_vec.size(); i++){
+    int block_size = (num_candidates_vec[i] + thread_size - 1) / thread_size;
+    const float *loc_data = loc_data_vec[i];
+    const float *obj_data = obj_data_vec[i];
+    const float *cls_data = cls_data_vec[i];
+    const float *ori_data = ori_data_vec[i];
+    const float *dim_data = dim_data_vec[i];
+    const float *anchor_data = yolo_blobs.anchor_blob->gpu_data()
+                               + num_anchor_per_scale * 2 * i;
```

Comment:
static_cast<float>(num_anchor_per_scale * 2 * i)

Reply:
This is an index, so integer should be used here.